drafts/bdd-specflow.md

```diff
@@ -0,0 +1,191 @@
+publishAs: alexander-duschek
+hideFromHashnodeCommunity: false
+ignorePost: true
+---
+
+Behaviour-Driven Development (BDD) is a technique of agile software development to ensure that the requirements of the software to be developed are fulfilled. To achieve this a special syntax called Gherkin (see below) is used to write down the specification which then can be translated by tools into test case skeletons. The developers add the test code and write/enhance the application until the test passes.
+A big advantage is that stakeholders, customers, business analysts, requirement engineers, all parties of a project can provide tests.
+
+## BDD Key features
+
+- Everyone can provide test cases
+- Test cases are easy to read
+- Good tool support for automation
+- Developers need less domain knowledge
+- Developers can focus on writing tests and features instead of finding test cases
+- Project documentation on the fly
+
+## What about Test-Driven Development (TDD)?
+
+Simply put, if a software developer uses the TDD technique, he first creates unit tests based on the requirements or the specification of the software or feature. Afterward, he enhances the software until the tests will eventually pass. For more details read the following article.
+
+https://medium.com/geekculture/test-driven-development-that-every-developer-should-know-about-with-example-c5f68af547a6
+
+As you probably noticed, BDD and TDD can be combined. The main difference is that with TDD the developer creates the test cases and with BDD every person related to the project can supply tests with the common Gherkin syntax (see next paragraph).
+
+## Gherkin syntax
+The Gherkin syntax is a way to describe can be seen in the image below. It consists of keywords at the beginning of each line followed by explaining text.
+
+```
+Feature: Login
+    Ensures that access to internal areas is only provided for authenticated users
+    
+Scenario: Invalid login credentials prevent access to internal areas
+    Given the user has opened the login page
+    When the user enters invalid login credentials
+    When the user clicks on the login button
+    Then an error message is shown
+    Then the user stays on the login page
+    
+Scenario: Valid login credentials allow access to internal areas
+    Given the user has opened the login page
+    When the user enters valid login credentials
+    When the user clicks on the login button
+    Then the user is redirected to an internal page
+```
+
+Feature describes a high-level software feature and is used to group Scenarios. A Scenario is a test case for the feature. It consists of preconditions (Given), actions (When), and assertions (Then). There are many possibilities to provide more detailed information but the basic set of a feature file looks like this in Gherkin syntax.
+
+To make it more natural, identical keywords can be chained by and. So instead of
```

Comment:
I'd highlight "and" somehow, with double quote, italics or inline code. Otherwise it looks like an unfinished sentence at first :D